### DISCUSSION TITLE
Airflow modules (worker, scheduler and webserver) are unable to connect to PG via pg_bouncer

### DISCUSSION THREAD

**Discussion by akshay-gupta-meesho** [apache/airflow#28598] [General]
### Official Helm Chart version

1.7.0 (latest released)

### Apache Airflow version

2.4.3

### Kubernetes Version

1.22

### Helm Chart configuration

```
data:
  metadataConnection:
    user: xxxx
    pass: xxxxx
    protocol: postgresql
    host: xxxxx
    port: 5432
    db: xxxxx
    sslmode: disable

  brokerUrl: xxxx
```

### What happened

PG Bouncer is able to connect to our internal DB but rest of the airflow modules i.e., webserver, woker ans scheduler are trying to connect to PG Bouncer, they are failing at init container (wait-for-airflow-migrations) initialisation with following error - 
```
....................
ERROR! Maximum number of retries (20) reached.

Last check result:
$ airflow db check
Traceback (most recent call last):
  File "/home/airflow/.local/bin/airflow", line 5, in <module>
    from airflow.__main__ import main
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/__init__.py", line 34, in <module>
    from airflow import settings
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/settings.py", line 37, in <module>
    from airflow.configuration import AIRFLOW_HOME, WEBSERVER_CONFIG, conf  # NOQA F401
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/configuration.py", line 1734, in <module>
    conf.validate()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/configuration.py", line 339, in validate
    self._validate_config_dependencies()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/configuration.py", line 441, in _validate_config_dependencies
    raise AirflowConfigException(f"error: cannot use sqlite with the {self.get('core', 'executor')}")
airflow.exceptions.AirflowConfigException: error: cannot use sqlite with the CeleryExecutor

```

<img width="996" alt="image" src="https://user-images.githubusercontent.com/108790221/209545850-ef6153aa-2b53-415d-80a9-255420357280.png">

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Reply by potiuk** [reply]
This is entirely unproductive to post entire configuration of yours. You seem to have problem with configuration of yours and unless you specify the exact values you configured and changed vs. the original values, don't expect anyone to do it for you to help you with resolving your configuration issue. You are asking someone who would attempt to help you to do the job of really finding out what you have changed.  This is asking a lot from other people in order to help you with solving potential typos and misconfiguraitons you've made.

I suggest you spend a little of your time to extract only the changes you've made to be little more mindful to time of people here who try to help you in their free time.

In any way this is a discussion, not an issue, so I convert it to discussion.

**Reply by potiuk** [reply]
And please avoid posting troubleshooting of your configuration as issues in the future - open discussion instead.

**Reply by akshay-gupta-meesho** [reply]
Thanks for the suggestion.

Updated helm values.conf values that I have changed.

**Reply by potiuk** [reply]
You might have some other custom modifications that apparently prevent the container to get the postgres configuration.. I suspect you have some - also you seem to missed the "pgbouncer.enabled = Trye" which you should have enabled to get pgbouncer up and running. So I guess there are other changes and it's not possible to help you to solve your problem if you won't tell us all the differences you applied

Can you is the generated resource for "wait-for-airflow-migration" containers when you add ``--dry-run`` parameter on help and make sure that you review and compare (and provide  here ALL modifications you've made vs. standard configuration).
